Update VReplication Reference Pages

mattlord opened the pull request:
Documents: https://github.com/vitessio/vitess/pull/11705 (likely some stragglers to be corrected in follow-ups as we review the full set of docs)

Fixes: https://github.com/vitessio/website/issues/1242

mattlord committed:
Remove vrepl v1 commands in v16 and add removal warning in v15-

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
Remove more vrepl v1 command content

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
MoveTables page

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
Reshard page

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
Conslidate MoveTables docs

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
Consolidate Reshard page

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
Update materialize

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
Another round of changes

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
Another round

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
Add Complete action to MoveTables and Reshard

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
Syncing up with https://github.com/vitessio/website/pull/1187

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord commented on content/en/docs/16.0/user-guides/migration/cutover.md:
```diff
@@ -168,9 +65,9 @@ Depending on the size of the table(s), the VDiff process may need increased time
-Note that running VDiff via vtctld can lead to vtctld consuming significantly more memory than usual. 
-We've found this to be around 1 GB plus, instead of the a few hundred MB that it normally uses. 
-If you run your vtctld in a memory limited container, you may want to take this into account. 
+Note that running VDiff via vtctld can lead to vtctld consuming significantly more memory than usual.
```
Copying this relevant comment over: https://github.com/vitessio/website/pull/1187/files#r1002133833

rohit-nayak-ps commented on content/en/docs/16.0/reference/vreplication/metrics.md:
```diff
@@ -4,13 +4,13 @@ description: Metrics related to vreplication functionality
 ---
 
-VReplication exports several metrics using the expvars interface. These are available at the `debug/vars` endpoint of vttablet's http status pages. [More details here](../../features/monitoring/#3-push-based-metrics-system#3-push-based-metrics-system)
+VReplication exports several metrics using the expvars interface. These are available at the `/debug/vars` endpoint of vttablet's http status pages. [More details can be found here](../../features/monitoring/#3-push-based-metrics-system#3-push-based-metrics-system).
```
The anchor is repeated and hence link doesn't work as expected.

rohit-nayak-ps commented on content/en/docs/16.0/reference/vreplication/movetables.md:
```diff
@@ -246,48 +236,176 @@ We caution against against using `EXEC` or `EXEC_IGNORE` for the following reaso
+
+###### Uses
+* If you just want a consistent snapshot of all the tables you can set this flag. The workflow
+will stop once the copy is done and you can then mark the workflow as `Complete`d
```
`Complete`d doesn't render well, probably best to remove the trailingd.

rohit-nayak-ps commented on content/en/docs/16.0/reference/vreplication/movetables.md:
```diff
@@ -246,48 +236,176 @@ We caution against against using `EXEC` or `EXEC_IGNORE` for the following reaso
+
+</div>
+
+###### Uses
```
Incorrect cut/paste: this is the same text as for `auto_start`

rohit-nayak-ps commented on content/en/docs/16.0/reference/vreplication/reshard.md:
```diff
@@ -198,28 +226,111 @@ We caution against against using `EXEC` or `EXEC_IGNORE` for the following reaso
+</div>
+
+###### Uses
+* allows updating the rows in `_vt.vreplication` after `Reshard` has setup the
```
Same issue here as in MoveTables. Need to change this text.

rohit-nayak-ps reviewed (changes requested):
Nice! 

Noticed a few minor issues. Otherwise looks very good.

mattlord committed:
Merge remote-tracking branch 'origin/prod' into vrepl_reference

mattlord committed:
Address review comments

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord commented:
Thanks, @rohit-nayak-ps ! I believe that I addressed your comments here: https://github.com/vitessio/website/pull/1267/commits/e02d88c52b943f2f15e87fb90e8516958c1baa1e

mattlord committed:
Add VTGate VStream info to tablet selection doc

Signed-off-by: Matt Lord <mattalord@gmail.com>

rohit-nayak-ps commented on content/en/docs/16.0/reference/vreplication/tablet_selection.md:
```diff
@@ -1,24 +1,28 @@
 ### Introduction
 
-For [VReplication streams](../../../concepts/vstream/), we must choose a tablet to serve the role of source (vstreamer) and target (vapplier) in the replication stream and this is done automatically.
+For both [VTGate VStreams and VTTablet VReplication streams](../../../concepts/vstream/) we must choose a tablet to serve the role of source (vstreamer). For VReplication streams we must also choose a a tablet to serve the role of target (vapplier). This tablet selection is performed by the internal `TabletPicker` component.
```
We don't choose a target tablet: vreplication streams are running on the target shard's primary.

mattlord replied:
Good catch! Mistake in modifying that paragraph. I've cleaned it up here: https://github.com/vitessio/website/pull/1267/commits/e4696ec3fbe3ac4ace100d4ce919331cc0535af3

rohit-nayak-ps commented on content/en/docs/16.0/reference/vreplication/tablet_selection.md:
```diff
@@ -1,24 +1,28 @@
 ### Cells
 
-VReplication will only look for source and target tablet pairings within the same cell by default — so if the target primary is in the `zone1` cell it will only look for source tablets in `zone1` cell. If you want to have cross-cell streams then you will need to specify the list of cells or any [CellAlias](https://vitess.io/docs/reference/programs/vtctl/cell-aliases/) that contain the list of pot ...
+The `TabletPicker` will only look for source and target tablet tablets within the same cell of the calling process by default — the `vtgate` managing the VStream or the target `vttablet` for a VReplication stream — so e.g. if the target primary tablet is in the `zone1` cell it will only look for source tablets in the `zone1` cell. If you want to have cross-cell streams then you will need to speci ...
```
typo: "tablet tablets"

mattlord committed:
Tweaks and corrections to tablet selection

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
One more...

Signed-off-by: Matt Lord <mattalord@gmail.com>

mattlord committed:
Moar minor tablet selection tweaks

Signed-off-by: Matt Lord <mattalord@gmail.com>
